### PR TITLE
fix(sdk): single task execute status bug again

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [x] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/core/model/model.py [hunk 1]
```diff
@@ -268,13 +269,19 @@ def eval_user_handler(
             else:
                 _scheduler.schedule_single_task(step, task_index)
         except Exception as e:
-            _manifest["status"] = STATUS.FAILED
+            _status = STATUS.FAILED
```
Comment (tianweidut) [inline client/starwhale/core/model/model.py:272]: `_status=` in line 256 and line 272 is useless.
`_status` in finally code block will overwrite it.

File: client/starwhale/core/model/model.py [hunk 2]
```diff
@@ -268,13 +269,19 @@ def eval_user_handler(
...
         finally:
-            _status = True
-            for _step in _steps:
-                _status = _status and _step.status == STATUS.SUCCESS
+            _status = (
+                STATUS.SUCCESS
+                if all(
+                    _s.status == STATUS.SUCCESS
+                    for _s in _steps
+                    if (step and _s.step_name == step) or not step
```
Comment (tianweidut) [inline client/starwhale/core/model/model.py:281]: rename step -> step_name ?

Comment (tianweidut) [reply client/starwhale/core/model/model.py:281]: `if (step and _s.step_name == step) or not step` -> `if step == "" or step == _s.step_name` ?

File: client/starwhale/core/model/model.py (hunk 1 above)
Comment (goldenxinxing) [reply client/starwhale/core/model/model.py]: yes, already change to try block

**Review by tianweidut — APPROVED** [review]

**Pull request merged**